### PR TITLE
fix(aws-lambda,netlify-lambda):  binary body v2 and cookies v1

### PR DESCRIPTION
**Pull request by Hebilicious**
### 🔗 Linked issue

Fix https://github.com/unjs/nitro/issues/852

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The aws-lambda preset works with apigw v1 and v2. 
This PR fix the body type for apigw v2, and the cookies for apigw v1

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/utils.lambda.ts [hunk 1]
```diff
@@ -31,18 +31,18 @@ export function normalizeLambdaOutgoingHeaders(
 export async function normalizeLambdaOutgoingBody(
   body: BodyInit | ReadableStream | Buffer | Readable | Uint8Array,
   headers: Record<string, number | string | string[] | undefined>
-): Promise<{ type: "text" | "binary"; body: string }> {
+): Promise<{ bodyType: "text" | "binary"; body: string }> {
```
Comment (pi0) [inline src/runtime/utils.lambda.ts:34]: Was this change necessary?

Comment (pi0) [reply src/runtime/utils.lambda.ts:34]: I mainly refactored like this to avoid restructuring and having simple keys. I would appreciate if keep changes in PR minimal 🙏🏼 (one idea if you don't like long names, we can call variable `_body` or `awsBody` instead of `normalizedBody`.)

Comment (Hebilicious) [reply src/runtime/utils.lambda.ts]: Got it, updated everything to `awsBody` 👍🏽

File: src/runtime/entries/netlify-lambda.ts [hunk 2]
```diff
@@ -36,15 +36,15 @@ export async function lambda(
...
     headers: normalizeLambdaOutgoingHeaders(r.headers, true),
-    body: outBody.body,
-    isBase64Encoded: outBody.type === "binary",
-    multiValueHeaders: {
-      ...(cookies.length > 0 ? { "set-cookie": cookies } : {}),
-    },
+    body: awsBody.body,
+    isBase64Encoded: awsBody.type === "binary",
+    ...(cookies.length > 0 && {
+      multiValueHeaders: { "set-cookie": cookies },
```
Comment (pi0) [inline src/runtime/entries/netlify-lambda.ts:47]: Just checking, was this (another) fix or refactor to avoid empty object?

File: test/presets/aws-lambda.test.ts [hunk 3]
```diff
@@ -62,21 +57,23 @@ describe("nitro:preset:aws-lambda", async () => {
...
-        status: res.statusCode,
-        headers: resHeaders,
-      };
+      return makeResponse(res, "v2");
     };
   });
 });
+
+const makeResponse = (response: any, version: "v1" | "v2") => {
+  const headers = { ...response.headers };
```
Comment (pi0) [inline test/presets/aws-lambda.test.ts:66]: Perf: We don't need to duplicate memory for headers if there is no `set-cookie` in it (nor the version checks logic). I wouldn't also make a copy even for `set-cookie` unless it is necessary to prevent a valid scenario where modifying original headers makes issues

**Comment by pi0** [comment]
@Hebilicious Since planning for next patch release, what is blocker for this PR? Do you need to verify it against a deployment?

File: src/runtime/entries/netlify-lambda.ts (hunk 2 above)
Comment (Hebilicious) [reply src/runtime/entries/netlify-lambda.ts:47]: Refactor due to the initial fix, I made the modification here too so that both `aws-lambda` and `netlify-lambda` have consistent behaviour.

**Comment by Hebilicious** [comment]
> @Hebilicious Since planning for next patch release, what is blocker for this PR? Do you need to verify it against a deployment?

Was just in draft because WIP, it should be good to go now 👍🏽

**Comment by EnzoAlbornoz** [comment]
It would be great if this is merged. I'm having issues with hosting IPX on my lambda function.

**Review by pi0 — APPROVED** [review]
Thanks (i haven't test it through but i trust you did)
